board/app/controllers/thread_controller.php

```diff
@@ -1,107 +1,141 @@
-    }   
+    }  
 
-    //Displays the comments of the thread
-    public function view() 
+    public function mythreads()
     {
-        $per_page = self::MAX_COMMENT_PER_PAGE;
+        $per_page = self::MAX_THREAD_PER_PAGE;
         $current_page = Param::get('page', 1);
-        $pagination = new SimplePagination($current_page, $per_page) ;
+        $pagination = new SimplePagination($current_page, $per_page);
 
-        $thread = Thread::get(Param::get('thread_id'));
-        $comments = $thread->getComments($pagination->start_index -1, $pagination->count + 1);
-        
-        $pagination->checkLastPage($comments);
-        $total = $thread->countComments();
+        $id = $_SESSION['user_id'];
+        $threads = Thread::getAllMyThread($pagination->start_index -1, $pagination->count + 1, $id);
+        $pagination->checkLastPage($threads);
+
+        $total = Thread::countAllThreadByUserId($_SESSION['user_id']);
         $pages = ceil($total / $per_page);
+
         $this->set(get_defined_vars());
+        $this->render('my_threads');
     }
 
-    //Write a new comment
-    public function write()
+    public function bycategory() 
     {
-        $thread = Thread::get(Param::get('thread_id'));
-        $comment = new Comment();
-        $current_page = Param::get('page_next', 'write');
+        $category = Param::get('category','none');
+            if ( $category !== 'none') {
+                $per_page = self::MAX_THREAD_PER_PAGE; 
+                $current_page = Param::get('page', 1);
+                $pagination = new SimplePagination($current_page, $per_page);
 
-        switch($current_page) { 
-            case 'write':
-            break;
+                $threads = Thread::getByCategory($pagination->start_index -1, $pagination->count + 1, $category);
+                $pagination->checkLastPage($threads);
 
-        case 'write_end':                
-            $comments->username = Param::get('username');
-            $comment->body = Param::get('body');
-          
-            try 
-            {            
-                $thread->write($comment);
-            } catch (ValidationException $e) {                    
-                $current_page = 'write';
-            }                        
+                $total = Thread::countAllThreadByCategory($category);
+                $pages = ceil($total / $per_page);
+                $this->set(get_defined_vars());
+                $this->render('under_category');
         
-        break;
-        default:
-            throw new NotFoundException("{$current_page} is not found");
-            break;
-        }
-
-    $this->set(get_defined_vars());
-    $this->render($current_page);   
-    }
-}
-?>
+            } else {
+                $categories = Thread::getAllCategory();
+                $this->set(get_defined_vars());
+                $this->render('categories');
+            }
+        }    
+
+    public function under_category() 
```

Comment:
done po. :bow"
Changed it po to sub_category.
--> https://github.com/maragatchalian/FoodFacts/commit/7cc4e579dd177ec3f0f261b07213efeacba43d1b
:blush: